scripts/dataset_processing/process_speech_commands_data.py

```diff
@@ -76,135 +83,191 @@ def extract_file(filepath: str, data_dir: str):
-    data_folder: str, dst_folder: str, rebalance: bool = False, class_split: str = "all", skip_duration: bool = False
-):
-    """
-    To generate manifest
+def __construct_filepaths(
+    all_files: List[str],
+    valset_uids: Set[str],
+    testset_uids: Set[str],
+    class_split: str,
+    class_subset: List[str],
+    pattern: str
+) -> Tuple[Dict[str, int], Dict[str, List[tuple]], List[tuple], List[tuple], List[tuple], List[tuple], List[tuple]]:
 
-    Args:
-        data_folder: source with wav files and validation / test lists
-        dst_folder: where manifest files will be stored
-        rebalance: rebalance the classes to have same number of samples.
-        class_split: whether to use all classes as distinct labels, or to use
-            10 classes subset and rest of the classes as noise or background.
-        skip_duration: Bool whether to skip duration computation. Use this only for
-            colab notebooks where knowing duration is not necessary for demonstration.
+    label_count = defaultdict(int)
+    label_filepaths = defaultdict(list)
+    unknown_val_filepaths = []
+    unknown_test_filepaths = []
 
-    Returns:
+    train, val, test = [], [], []
+    for entry in all_files:
+        r = re.match(pattern, entry)
+        if r:
+            label, uid = r.group(2), r.group(3)
 
-    """
+            if label == '_background_noise_' or label == 'silence':
+                continue
 
-    if not os.path.exists(dst_folder):
-        os.makedirs(dst_folder)
+        if class_split == 'sub' and label not in class_subset:
+            label = 'unknown'
 
-    # Used for 10 classes + silence + unknown class setup
-    class_subset = ["yes", "no", "up", "down", "left", "right", "on", "off", "stop", "go"]
+            if uid in valset_uids:
+                unknown_val_filepaths.append((label, entry))
+            elif uid in testset_uids:
+                unknown_test_filepaths.append((label, entry))
 
-    pattern = re.compile(r"(.+\/)?(\w+)\/([^_]+)_.+wav")
-    all_files = glob.glob(os.path.join(data_folder, '*/*wav'))
+        if uid not in valset_uids and uid not in testset_uids:
+            label_count[label] += 1
+            label_filepaths[label].append((label, entry))
 
-    with open(os.path.join(data_folder, 'validation_list.txt'), 'r') as fin:
-        validation_files = fin.readlines()
+        if label == 'unknown':
+            continue
 
-    valset = set()
-    for entry in validation_files:
-        r = re.match(pattern, entry)
-        if r:
-            valset.add(r.group(3))
+        if uid in valset_uids:
+            val.append((label, entry))
+        elif uid in testset_uids:
+            test.append((label, entry))
+        else:
+            train.append((label, entry))
 
-    with open(os.path.join(data_folder, 'testing_list.txt'), 'r') as fin:
-        testing_files = fin.readlines()
+    return label_count, label_filepaths, unknown_val_filepaths, unknown_test_filepaths, train, val, test
```

Comment:
Can you return a names tuple / data class with these results? This is quite messy.